src/ui/bind_handlers.js

```diff
@@ -178,7 +178,11 @@ export default function bindHandlers(map: Map, options: {interactive: boolean, c
             contextMenuEvent = e;
         }
 
-        e.preventDefault();
+        // prevent browser context menu when necessary; we don't allow it with rotation
+        // because we can't discern rotation gesture start from contextmenu on Mac
+        if (map.dragRotate.isEnabled() || map.listens('contextmenu')) {
```

Comment:
I understand the first part of this conditional but am I thinking about it correctly that `map.listens('contextmenu')` would be `true` if a `contextmenu` listener has been set up on the map? Why would having a listener prevent the event? It seems like odd behavior that I could disable drag rotate, set up a listener for `contextmenu` and then the context menu wouldn't actually open.

Also, the docs say that `listens` should return `false` if no listener is set up, but I actually get `undefined`.

Reply:
The original reason for having Map `contextmenu` event is to allow specifying custom behavior on right click, such as opening your own HTML-styled app-specific context menu, and in this case it doesn't make sense to also show the browser default one. 

Leaflet [does the same thing](https://github.com/Leaflet/Leaflet/blob/9fda888df2791ca0df6f182a9d91d393ee49034c/src/map/Map.js#L1369) and no one raised concerns about it yet, so I think it's safe to do the same logic here.

Reply:
Ah, ok that makes sense. I tested this on Mac with Safari, Firefox and Chrome and they all worked as expected so I think this is good.

Reply:
This is not working in ios WKWebView.